### PR TITLE
Enable `one-key-formatting` on title fields

### LINKED ISSUE #3718: Make `one-key-formatting` also works in any title field
Now that `parse-backticks` feature works in issue titles, commit titles and more places, why not make `one-key-formatting` also works in any title field? I think many RGH users who enable `parse-backticks` feature have get used to adding backticks when writing a title. It would be a great convenience.
IMO, not all the symbols supported in comment field also need to be supported in title textarea. Maybe supporting ``[ ` ' "`` is enough.

BTW, it seems that `one-key-formatting` doesn't work consistently. But I don't know how to reproduce it. Just encouter this bug again and again🤣.

URL: issue page, pull request page and more

### PR DESCRIPTION
**Pull request by yakov116**
Closes #3718

## Test URLs
⏫ ⬆️ ⏫ ⬆️ ⏫ ⬆️ ⏫ ⬆️ ⏫ ⬆️ ⏫ ⬆️ ⏫ ⬆️ ⏫ ⬆️ ⏫ ⬆️ ⏫ ⬆️ 

## Screenshot
None

Need some typescript help

Also it was simpler to just enable all format options. Do you want to limit it?

### CODE REVIEW

**Comment by fregante** [comment]
> Also it was simpler to just enable all format options

👍 

> Need some typescript help

I'll try

File: source/github-events/on-field-keydown.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const onCommentFieldKeydown = (callback: delegate.EventHandler<KeyboardEvent, HTMLTextAreaElement>) => {
+	onFieldKeydown('.js-comment-field, #commit-description-textarea', callback);
+};
+
+const onTitleFieldKeydown = (callback: delegate.EventHandler<KeyboardEvent, HTMLInputElement>) => {
+	onFieldKeydown('#issue_title', callback);
+};
+
+export {onCommentFieldKeydown, onTitleFieldKeydown};
```
Comment (fregante) [inline source/github-events/on-field-keydown.tsx:28]:
```suggestion
export function onCommentFieldKeydown (callback: delegate.EventHandler<KeyboardEvent, HTMLTextAreaElement>): void {
	onFieldKeydown('.js-comment-field, #commit-description-textarea', callback);
};

export function onTitleFieldKeydown (callback: delegate.EventHandler<KeyboardEvent, HTMLInputElement>): void {
	onFieldKeydown('#issue_title', callback);
};
```

File: source/github-events/on-field-keydown.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import select from 'select-dom';
+import delegate from 'delegate-it';
+
+type DelegateFieldEvent = delegate.EventHandler<KeyboardEvent, HTMLTextAreaElement>;
```
Comment (fregante) [inline source/github-events/on-field-keydown.tsx:4]: Not the correct solution, sadly, but thankfully there's no difference between `textarea` and `input[type=text]`

**Review by fregante — APPROVED** [review]
WFM

2gif

**Comment by fregante** [comment]
However… maybe I'd like this to be refactored once again to:

- allow more input fields, like the commit title field
- remove `onTitleFieldKeydown` because it's too specific and the selector is too simple

File: source/github-events/on-field-keydown.tsx (hunk 2 above)
Comment (yakov116) [reply source/github-events/on-field-keydown.tsx:4]:
```suggestion
type DelegateFieldEvent = delegate.EventHandler<KeyboardEvent, HTMLTextAreaElement>; // For testing
```

Comment (fregante) [reply source/github-events/on-field-keydown.tsx:4]: What testing?

File: source/github-events/on-field-keydown.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+function onFieldKeydown(selector: string, callback: DelegateFieldEvent): void {
+	delegate<HTMLTextAreaElement, KeyboardEvent>(document, selector, 'keydown', event => {
+		const field = event.delegateTarget;
+
+		// Don't do anything if the autocomplete helper is shown or if non-latin input is being used
```
Comment (fregante) [inline source/github-events/on-field-keydown.tsx:10]:
```suggestion
		// The suggester is GitHub’s autocomplete dropdown
```

File: source/github-events/on-field-keydown.tsx (hunk 2 above)
Comment (yakov116) [reply source/github-events/on-field-keydown.tsx:4]: That the commit title works (Commit suggestion)

**Review by fregante — APPROVED** [review]
Seems ok

File: source/github-events/on-field-keydown.tsx (hunk 3 above)
Comment (yakov116) [reply source/github-events/on-field-keydown.tsx:10]: 6d852ce1d75bf9ce59bedb902300d9d5a2a8b4e6

Comment (yakov116) [reply source/github-events/on-field-keydown.tsx:10]: @fregante do you still want the wording changed?

**Pull request merged**